src/test/test_parallel_pipeline.cpp

```diff
@@ -647,17 +651,22 @@ int TestMain() {
 
         // Run test several times with different types
         run_function_spec();
-        run_function<size_t,int>("size_t", "int");
-        run_function<int,double>("int", "double");
-        run_function<size_t,double>("size_t", "double");
-        run_function<size_t,bool>("size_t", "bool");
-        run_function<int,int>("int","int");
-        run_function<check_type<unsigned int>,size_t>("check_type<unsigned int>", "size_t");
-        run_function<check_type<unsigned short>,size_t>("check_type<unsigned short>", "size_t");
-        run_function<check_type<unsigned int>, check_type<unsigned int> >("check_type<unsigned int>", "check_type<unsigned int>");
-        run_function<check_type<unsigned int>, check_type<unsigned short> >("check_type<unsigned int>", "check_type<unsigned short>");
-        run_function<check_type<unsigned short>, check_type<unsigned short> >("check_type<unsigned short>", "check_type<unsigned short>");
-        run_function<double, check_type<unsigned short> >("double", "check_type<unsigned short>");
+        #define M(type1, type2) run_function<type1, type2>(#type1, #type2);
```

Comment:
I suggest to rename the macro to `TEST_WITH`.

Reply:
Then I would suggest RUN_FUNCTION. Less is more. More less is better than less less. :-)